src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -284,6 +305,49 @@ static bool ShouldBeFilteredOutOfExpandedCompletionList(VSCompletionItem item, I
                 // 2. or, all associated expanders are unselected, therefore should be excluded
                 return associatedWithUnselectedExpander;
             }
+
+            static ImmutableArray<CompletionItemWithHighlight> GetHighlightedList(
+                ImmutableArray<ExtendedFilterResult> filterResults,
+                string filterText,
+                CompletionHelper completionHelper,
+                bool highlightMatchingPortions)
+            {
+                if (!highlightMatchingPortions)
+                {
+                    return filterResults.SelectAsArray(r => new CompletionItemWithHighlight(r.VSCompletionItem, ImmutableArray<Span>.Empty));
+                }
+
+                var highlightedItems = ArrayBuilder<CompletionItemWithHighlight>.GetInstance(filterResults.Length);
+                foreach (var extendedResult in filterResults)
+                {
+                    ImmutableArray<Span> highlightedSpans;
+                    var filterResult = extendedResult.FilterResult;
+
+                    if (filterResult.CompletionItem.HasDifferentFilterText)
+                    {
+                        // The PatternMatch in FilterResult is calculated based on Roslyn item's FilterText, 
+                        // which can be used to calculate highlighted span for VSCompletion item's DisplayText w/o doing the matching again.
+                        // However, if the Roslyn item's FilterText is different from its DisplayText,
+                        // we need to do the match against the display text of the VS item directly to get the highlighted spans.
+                        highlightedSpans = completionHelper.GetHighlightedSpans(extendedResult.VSCompletionItem.DisplayText, filterText, CultureInfo.CurrentCulture).SelectAsArray(s => s.ToSpan());
+                    }
+                    else if (filterResult.PatternMatch.HasValue)
+                    {
+                        // Since VS item's display text is created as Prefix + DisplayText + Suffix, we can calculate the highlighted span by adding an offset that is the length of the Prefix.
+                        highlightedSpans = filterResult.PatternMatch.Value.MatchedSpans.SelectAsArray(s => s.ToSpan(filterResult.CompletionItem.DisplayTextPrefix?.Length ?? 0));
```

Comment:
feels like we'd only want to do this based on the item having a prefix.

Reply:
I think that is what it is doing?

Reply:
basically, you're always make a copy of hte array.  i'm saying: don't make the copy **unless** `filterResult.CompletionItem.DisplayTextPrefix.Length  > 0`

Reply:
It's not a copy of the array, we are converting an array of `TextSpan` to `Span`.

Reply:
ah, i missed that.